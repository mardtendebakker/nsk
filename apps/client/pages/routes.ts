export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
