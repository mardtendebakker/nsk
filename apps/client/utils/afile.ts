import { AFile } from './axios/models/aFile';

export function buildAFileLink(afile: AFile): string {
  return `https://${afile.unique_server_filename}.s3.amazonaws.com/${afile.discr}/${afile.original_client_filename}`;
}
