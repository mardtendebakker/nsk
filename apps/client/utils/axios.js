import axios from 'axios';

export { AxiosPromise, AxiosResponse, AxiosError } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333' : 'production_url';

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const SUPLIERS_PATH = '';

export default client;
