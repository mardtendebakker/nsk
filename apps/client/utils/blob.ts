export function openBlob(blob: Blob): void {
  const a = document.createElement('a');
  document.body.appendChild(a);
  a.style.cssText = 'display: none';

  const url = window.URL.createObjectURL(blob);

  a.href = url;
  a.target = '_blank';
  a.click();
  window.URL.revokeObjectURL(url);
}
