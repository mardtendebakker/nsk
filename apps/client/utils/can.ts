import _ from 'lodash';
import { Group } from '../stores/security/types';

export default function can(userGroups: Group[], requiredGroups: Group[], disableDefaultGroups = false): boolean {
  return _.intersection(userGroups, disableDefaultGroups ? requiredGroups : [...requiredGroups, 'admin', 'super_admin']).length > 0;
}
