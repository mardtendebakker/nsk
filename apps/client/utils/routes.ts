export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
export const CUSTOMERS_CONTACTS = '/customers/contacts';
export const CUSTOMERS_CONTACTS_NEW = '/customers/contacts/new';
export const CUSTOMERS_EMAILS_NEW = '/customers/emails/new';
export const CUSTOMERS_EMAILS = '/customers/emails';
export const SUPPLIERS = '/suppliers/:id';
export const ACCOUNT_VERIFICATION = '/account-verification';
export const SETTINGS = '/settings';
export const ORDERS_PURCHASES = '/orders/purchases';
export const ORDERS_SALES = '/orders/sales';
export const STOCKS_PRODUCTS = '/stock/products';
export const STOCKS_REPAIR_SERVICES = '/stock/repair-services';
