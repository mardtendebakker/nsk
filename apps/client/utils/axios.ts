import axios from 'axios';

export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333/api' : process.env.NX_AXIOS_BASE_URL;

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const CUSTOMERS_PATH = '/customers/:id';
export const SUPPLIERS_PATH = '/suppliers/:id';

export default client;

export interface Customer {
  id: number,
  name?: string,
  representative?: string,
  kvk_nr?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  is_partner?: number,
}

export interface Supplier {
  id: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  partner?: string,
}
