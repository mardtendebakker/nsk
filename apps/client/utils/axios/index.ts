import axios from 'axios';
import axiosClient from './client';
import refreshToken from './refreshTokenInterceptor';

export {
  CUSTOMERS_PATH,
  SUPPLIERS_PATH,
  SIGN_IN_PATH,
  REFRESH_TOKEN_PATH,
  USER_INFO_PATH,
  SIGN_UP_PATH,
  CONFIRM_ACCOUNT_PATH,
  CHANGE_PASSWORD_PATH,
  SEND_VERIFICATION_CODE_PATH,
  FORGOT_PASSWORD_PATH,
  SALES_ORDERS_PATH,
  PURCHASE_ORDERS_PATH,
  STOCK_PRODUCTS_PATH,
  USERS_PATH,
  STOCK_REPAIR_SERVICES_PATH,
  ORDER_STATUSES_PATH,
  COMPANIES_PATH,
  LOCATIONS_PATH,
  BULK_PRINT_SALES_PATH,
  BULK_PRINT_PURCHASES_PATH,
  ORDERS_ANALYTICS_PATH,
} from './paths';
export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

axiosClient.interceptors.response.use((response) => response, refreshToken);

export const { CancelToken } = axios;

export default axiosClient;
