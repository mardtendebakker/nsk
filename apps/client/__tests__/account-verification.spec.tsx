import { fireEvent, render } from '@testing-library/react';
import AccountVerification from '../pages/account-verification';
import { SIGN_IN } from '../utils/routes';

const mockRouter = {
  push: jest.fn(),
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

const mockUseSecurity = {
  state: { get user() { return { emailVerified: false }; }, get loading() { return false; } },
  confirmAccount: jest.fn(),
  sendVerificationCode: jest.fn(),
  signOut: jest.fn(),
};

jest.mock('../hooks/useSecurity', () => jest.fn(() => mockUseSecurity));

describe('account-verification', () => {
  afterEach(() => {
    jest.spyOn(mockUseSecurity.state, 'loading', 'get').mockReset();
    jest.spyOn(mockUseSecurity.state, 'user', 'get').mockReset();
    mockUseSecurity.sendVerificationCode.mockReset();
    mockUseSecurity.confirmAccount.mockReset();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<AccountVerification />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should redirect to sign in page', () => {
    jest.spyOn(mockUseSecurity.state, 'user', 'get').mockReturnValue(null);
    const { queryByText } = render(<AccountVerification />);
    expect(mockRouter.push).toBeCalledWith(SIGN_IN);
    expect(queryByText('accountVerification')).toBeNull();
  });

  it('should redirect to home page', () => {
    jest.spyOn(mockUseSecurity.state, 'user', 'get').mockReturnValue({ emailVerified: true });
    const { queryByText } = render(<AccountVerification />);
    expect(mockRouter.push).toBeCalledWith(SIGN_IN);
    expect(queryByText('accountVerification')).toBeNull();
  });

  it('should not request account confirmation / invalid code', () => {
    const { container } = render(<AccountVerification />);

    const form = container.querySelector('form');
    fireEvent.submit(form);

    expect(mockUseSecurity.confirmAccount).not.toBeCalled();
  });

  it('should not request account confirmation / loading', () => {
    jest.spyOn(mockUseSecurity.state, 'loading', 'get').mockReturnValue(true);
    const { container } = render(<AccountVerification />);

    const input = container.querySelector("input[name='code']");
    fireEvent.change(input, { target: { value: 'code1' } });

    const form = container.querySelector('form');
    fireEvent.submit(form);

    expect(mockUseSecurity.confirmAccount).not.toBeCalled();
  });

  it('should request account confirmation', () => {
    const { container, getByLabelText } = render(<AccountVerification />);

    const input = getByLabelText('code');
    fireEvent.change(input, { target: { value: 'code' } });

    const form = container.querySelector('form');
    fireEvent.submit(form);

    expect(mockUseSecurity.confirmAccount).toBeCalledWith({ code: 'code' });
  });

  it('should request to send verification code', () => {
    const { getByText } = render(<AccountVerification />);

    const button = getByText('sendCode');
    fireEvent.click(button);

    expect(mockUseSecurity.sendVerificationCode).toBeCalled();
  });

  it('should not request to send verification code / loading', () => {
    jest.spyOn(mockUseSecurity.state, 'loading', 'get').mockReturnValue(true);
    const { getByText } = render(<AccountVerification />);

    const button = getByText('sendCode');
    fireEvent.click(button);

    expect(mockUseSecurity.sendVerificationCode).not.toBeCalled();
  });

  it('should logout', () => {
    const { getByText } = render(<AccountVerification />);

    const button = getByText('logout');
    fireEvent.click(button);

    expect(mockUseSecurity.signOut).toBeCalled();
  });
});
