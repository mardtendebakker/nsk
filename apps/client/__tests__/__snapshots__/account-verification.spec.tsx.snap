// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account-verification matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiBox-root css-3pjj5b"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError css-a29xdt-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      logout
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1jjz4e9-MuiTypography-root"
    >
      accountVerificationMessage
    </h4>
    <form>
      <div
        class="MuiStack-root css-jfdv4h-MuiStack-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            code
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="code"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                sendCode
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  code
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-1ulybn8-MuiStack-root"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-fullWidth css-1evprfk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        confirm
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </main>
</DocumentFragment>
`;
